### PR TITLE
feat: add event recording in the admin dashboard ✌🏾

### LINKED ISSUE #68: Upload event recordings in the Admin Dashboard 🎥
### Description

After every Fam Friday, the ColorStack team uploads videos of the event sessions to YouTube! Then, it's linked in the Member Wiki. In the effort to consolidate things into one place, it would be great to be able to access event recordings from the Member Profile.

### Acceptance Criteria

An admin should be able to do the following in the Admin Dashboard:
- Go to the "Events" page.
- Click the context menu for a particular event row, which opens up a modal.
- In that modal, there should be a form that allows you to upload a link.
- When saving that form, it should save it to our database.

### Additional Context

This is the current events page in the Admin Dashboard:

Screenshot 2024-04-01 at 6 31 15 PM

### PR DESCRIPTION
**Pull request by Joyeleke**
## Description ✏️

Closes #68 

After every Fam Friday, the ColorStack team uploads videos of the event sessions to YouTube! This PR enables admins to add a link to an event within the admin dashboard.

**Tasks**

- Added functionality to add a link to event.
Screenshot 2024-04-08 at 12 05 02 PM

Screenshot 2024-04-08 at 12 08 15 PM

- Added DB migrations to add column "eventLink" to DB.
- Wrote queries to update link in DB.

## Type of Change 🐞

- [x] Feature - A non-breaking change which adds functionality.
- [ ] Fix - A non-breaking change which fixes an issue.
- [ ] Refactor - A change that neither fixes a bug nor adds a feature.
- [ ] Documentation - A change only to in-code or markdown documentation.
- [ ] Tests - A change that adds missing unit/integration tests.
- [ ] Chore - A change that is likely none of the above.

## Checklist ✅

- [x] I have done a self-review of my code.
- [x] I have manually tested my code (if applicable).
- [x] I have added/updated any relevant documentation (if applicable).

### CODE REVIEW

**Comment by Joyeleke** [comment]
Glad to finally get a PR in!

Couple questions here:

- What should the type of eventLink be? and how can we implement a validation function for the link? (I can do this using regex if it is always a youtube link)

- Also small issue - but does "add" sound worse/better than "upload"?

- Are we updating the events table in the admin dashboard?

File: apps/admin-dashboard/app/routes/_dashboard.events.$id.add.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+const { link } = CreateEventFormData.keyof().enum;
+
+function AddEventLinkForm() {
+  const { error, errors } = getActionErrors(useActionData<typeof action>());
+
+  return (
+    <RemixForm className="form" method="post">
+      <Form.Field error={errors.link} label="Link" labelFor={link} required>
+        <Input id={link} name={link} required />
```
Comment (ramiAbdou) [inline apps/admin-dashboard/app/routes/_dashboard.events.$id.add.tsx:100]: If the event already has an event recording, I think it'd be clearer that the value would override the existing value if we set the `defaultValue` to the existing link. Can we add that?

You can use the [`getEvent`](https://github.com/colorstackorg/oyster/blob/5d3b522e87614bf03b95a268dcefef95ae44905b/packages/core/src/modules/event/queries/get-event.ts#L8) method in the `loader`!

**Review by ramiAbdou — COMMENTED** [review]
This looks really great, @Joyeleke! Thank you so much for implementing! ❤️

Overall the PR looks really solid, just had some minor comments. To answer some of your questions:
- You already added validation by adding the field to the `Event` schema. You also used that validation in the `action`! We don't need to enforce that it's a YouTube link, but a valid URL should do (see my comments).
- I think "add" sounds better than "upload" here since it's just a link!
- Yup, the `events` table is the one we're looking to update!

Also, if you could review this [contribution](https://github.com/colorstackorg/oyster/blob/main/CONTRIBUTING.md#making-a-pull-request) guide and update the PR name that would be great! 🙂

**Comment by Joyeleke** [comment]
Hey @ramiAbdou. Thanks so much for the kind words and feedback! Honestly really enjoyed the process of contributing (maybe a little too much lol😅). 

Just have to figure out how to stay ahead while the JS ecosystem keeps deciding they need a new baby everyday😅

File: apps/admin-dashboard/app/routes/_dashboard.events.$id.add.tsx (hunk 1 above)
Comment (Joyeleke) [reply apps/admin-dashboard/app/routes/_dashboard.events.$id.add.tsx]: Made all other changes. This particular one could still use some work. Is there an option for a placeholder value on form? That may be better than what I currently have.

Screenshot 2024-04-08 at 11 00 26 PM

Comment (ramiAbdou) [reply apps/admin-dashboard/app/routes/_dashboard.events.$id.add.tsx]: So actually HTML inputs have this attribute called [`defaultValue`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement#defaultvalue) which will automatically load the value into the input when the form is loaded! Meaning if you don't change that input, it'll still retain that value.

Wherever possible, we'll try to use the web standards and this is a good example of that! HTML FTW!

**Comment by ramiAbdou** [comment]
> Just have to figure out how to stay ahead while the JS ecosystem keeps deciding they need a new baby everyday😅

Haha that's so real. There's so many new tools coming out all the time! In general, it's great to drown out the noise (unless there's some tangible value being added with a new tool).

**Pull request merged**